File: wordcount/calcapi.py
#/bin/python

import re


def string2Word(data):
    words = re.split('[\W0-9_]+', data)
    words.remove('')
    print "words = {}".format(words)

    # init the hash map
    wordMap = dict()
    for word in words:
        if word not in wordMap.keys():
            wordMap[word] = 1
        else:
            wordMap[word] = wordMap[word] + 1

    print "wordMap = {}".format(wordMap)
        
        
    

if __name__ == "__main__":
    print "Begint to Calc"
    filename = "./text.file"
    with open(filename, 'r') as fin:
        data = fin.read()
        print "data = {}".format(data) 

    string2Word(data)

File: wordcount/calcapi.go
package main

import (
	"log"
	"io/ioutil"
)


func main() {
    var filename string = "./text.file"
	data, err := ioutil.ReadFile(filename)
	if err != nil {
		log.Fatal(err)
	}
	log.Printf(string(data))
	//calcWord(data)

	var wordMap map[string]int = make(map[string]int, 0)
	string2Word(string(data), wordMap)

	for i,v := range wordMap {
		log.Printf("word = %s, count = %d", i, v)
	}
	
}


func string2Word(data string, wordMap  map[string]int) {		

    //var wordMap map[string]int = make(map[string]int, 0)

	startIndex := 0
	letterStart := false
	for i, v := range data {
        // Letter yes or no
		if (v >=65 && v <=90) || (v >= 97 && v<=122) {
			if !letterStart {
				letterStart = true
				startIndex = i
			}
		} else {
			if letterStart {
				letterStart = false
				wordMap[data[startIndex:i]]++	
			}
		}
	}
	
	if letterStart {
		wordMap[data[startIndex:]]++ 
	}

}
